src/a12n.ts

```diff
@@ -0,0 +1,27 @@
+import ketting from './ketting';
+import { LinkNotFound } from 'ketting';
+
+
+
+export async function addUserPrivilege(authenticatedAsHref: string, origin: string, resource: any): Promise<void> {
```

Comment:
The 3 parameters that we need are:

* principal
* resource
* privilege

I don't think `origin` needs to be sent separately, whatever calls this function can compose it with `new URL()` instead of doing this inside the function.

All 3 arguments are strings

Reply:
so, I'm still passing origin, in this last commit. Wondering how to resolve this... 
I'm not quite sure how to compose it